### ISSUE TITLE
[help] polygon.cut error: TypeError: Cannot set properties of undefined (setting 'prev')

### ISSUE THREAD

**Issue by Fauny** [alexbol99/flatten-js#175]
version: latest

code:
```js 
{
  let { point, segment, circle, arc, Polygon, multiline } = Flatten;

  // Create polygon from json
  let json = [
    [
      {
        ps: { x: 641.64, y: 118.32, name: "point" },
        pe: { x: 641.64, y: 151.74, name: "point" },
        name: "segment"
      },
      {
        ps: { x: 641.64, y: 151.74, name: "point" },
        pe: { x: 504.66, y: 151.74, name: "point" },
        name: "segment"
      },
      {
        ps: { x: 504.66, y: 151.74, name: "point" },
        pe: { x: 504.66, y: 118.32, name: "point" },
        name: "segment"
      },
      {
        ps: { x: 504.66, y: 118.32, name: "point" },
        pe: { x: 641.64, y: 118.32, name: "point" },
        name: "segment"
      }
    ]
  ];

  let polygon = new Polygon(json);
  let svg = polygon.svg();

  // Create Multiline
  let mlj = [
    {
      ps: { x: 576.48, y: 118.32, name: "point" },
      pe: { x: 576.48, y: 274.14, name: "point" },
      name: "segment"
    },
    {
      ps: { x: 576.48, y: 274.14, name: "point" },
      pe: { x: 641.64, y: 274.14, name: "point" },
      name: "segment"
    }
  ];
  let ml = multiline(mlj.map((s) => segment(s)));

  return polygon.cut(ml);
}
```

**Comment by Fauny** [comment]
@alexbol99 help me,  pls.

**Comment by alexbol99** [comment]
Dear @Fauny ,
I see a problem, but I don't have a fix yet.
Working on it

**Comment by alexbol99** [comment]
Bug fixed in v1.5.3
Now it can cut correctly.
I used polygon.toArray() to get both parts as polygon instances.

<img width="645" alt="image" src="https://github.com/alexbol99/flatten-js/assets/6965440/95040bf9-ff3b-4bd0-9ea1-8ca93533738d">

**Comment by Fauny** [comment]
Thank u !!
